src/kademlia/get_item.cpp

```diff
@@ -70,6 +70,15 @@ void get_item::got_data(bdecode_node const& v,
 		{
 			if (!m_data.assign(v, salt, seq, pk, sig))
 				return;
+
+			if (!m_put_requested)
```

Comment:
it's not obvious to me that this makes sense. Even if you want to put you may want to receive non-authoritative data.